### PR TITLE
Input files for offline TkCommissioner tool, 81X

### PR DESCRIPTION
**Pull request by vdutta**
Production of input root files for the offline TkCommissioner tool. These files contain the information from the tracker maps produced for offline SiStrip DQM indexed by detId for a given run and can be used to get a better understanding of SiStrip DQM issues at the module level. This is the 81X version of #14232.

### CODE REVIEW

File: DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc [hunk 1]
```diff
@@ -68,12 +70,19 @@ SiStripOfflineDQM::SiStripOfflineDQM(edm::ParameterSet const& pSet) : configPar_
...
   createSummary_           = configPar_.getUntrackedParameter<bool>("CreateSummary",false);
+  createTkInfoFile_        = configPar_.getUntrackedParameter<bool>("CreateTkInfoFile",false);
   inputFileName_           = configPar_.getUntrackedParameter<std::string>("InputFileName","");
   outputFileName_          = configPar_.getUntrackedParameter<std::string>("OutputFileName","");
   globalStatusFilling_     = configPar_.getUntrackedParameter<int>("GlobalStatusFilling", 1);
   printFaultyModuleList_   = configPar_.getUntrackedParameter<bool>("PrintFaultyModuleList", false);
 
   nEvents_  = 0;
+
+  tkinfoTree_ = 0;
```
Comment (dmitrijus) [inline DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc]: use nullptr, for clarity :)

File: DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc [hunk 2]
```diff
@@ -68,12 +70,19 @@ SiStripOfflineDQM::SiStripOfflineDQM(edm::ParameterSet const& pSet) : configPar_
...
+  edm::Service<TFileService> fs;
```
Comment (dmitrijus) [inline DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc]: Ideally, it should be inside if's scope below. Not strictly necessary, but it makes a bit more sense.

**Comment by dmitrijus** [comment]
+1
I've added a few comments to the commits

File: DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc [hunk 3]
```diff
@@ -201,14 +210,19 @@ void SiStripOfflineDQM::endRun(edm::Run const& run, edm::EventSetup const& eSetu
     if (create_tkmap) {
       std::vector<edm::ParameterSet> tkMapOptions = configPar_.getUntrackedParameter< std::vector<edm::ParameterSet> >("TkMapOptions" );
       if (actionExecutor_->readTkMapConfiguration(eSetup)) {
+        std::vector<std::string> map_names;
 	
 	for(std::vector<edm::ParameterSet>::iterator it = tkMapOptions.begin(); it != tkMapOptions.end(); ++it) {
 	  edm::ParameterSet tkMapPSet = *it;
 	  std::string map_type = it->getUntrackedParameter<std::string>("mapName","");
+          map_names.push_back(map_type);
```
Comment (dmitrijus) [inline DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc]: a mix of tabs and spaces for indentation

File: DQM/SiStripMonitorClient/src/SiStripTrackerMapCreator.cc [hunk 4]
```diff
@@ -609,3 +612,75 @@ uint16_t SiStripTrackerMapCreator::getDetectorFlagAndComment(DQMStore* dqm_store
...
+      edm::LogInfo("SiStripTrackerMapCreator::createInfoFile") << "SiStrip top level directory is " << dirname;
+    }
+    dqm_store->cd();
+
+    std::vector<TkHistoMap*> tkHMaps;
+
+    uint32_t nHists = map_names.size();
+
+    for(uint32_t ih = 0; ih < nHists; ++ih) {
+      tkHMaps.push_back(new TkHistoMap());
```
Comment (davidlange6) [inline DQM/SiStripMonitorClient/src/SiStripTrackerMapCreator.cc]: how are these pointers deleted
@vdutta

**Comment by fioriNTU** [comment]
@vdutta can you please try to clarify what David asks, we need this PR merged asap

File: DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc (hunk 2 above)
Comment (vdutta) [inline DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc]: Hi @dmitrijus do you mean just line 82 or 81 (with the switch to nullptr) as well?

Comment (dmitrijus) [inline DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc]: Just 82, something like `if (createTkInfoFile_){ edm::Service<TFileService> fs; tkinfoTree_ = fs->make<TTree>("TkDetIdInfo", ""); }`

For 81: pointers should never be assigned or compared to 0, nullptr should be used instead.

Comment (vdutta) [inline DQM/SiStripMonitorClient/src/SiStripOfflineDQM.cc]: Ok, clear now ... thanks!

**Comment by dmitrijus** [comment]
+1

**Comment by Dr15Jones** [comment]
@davidlt will this go away if the tests are rerun or does this problem require waiting longer for an IB with the new clang?

**Comment by davidlt** [comment]
@Dr15Jones needs 2300 May 18th IB for this to go away, which is not yet available.

**Pull request merged**